webrtc/RTCPeerConnection-helper.js

```diff
@@ -182,16 +182,27 @@ function exchangeIceCandidates(pc1, pc2) {
 }
 
 // Helper function for doing one round of offer/answer exchange
-// betweeen two local peer connections
-function doSignalingHandshake(localPc, remotePc) {
-  return localPc.createOffer()
-  .then(offer => Promise.all([
-    localPc.setLocalDescription(offer),
-    remotePc.setRemoteDescription(offer)]))
-  .then(() => remotePc.createAnswer())
-  .then(answer => Promise.all([
-    remotePc.setLocalDescription(answer),
-    localPc.setRemoteDescription(answer)]))
+// between two local peer connections
+async function doSignalingHandshake(localPc, remotePc, options={}) {
```

Comment:
Ok, but I dislike helpers options when they turn into replacement APIs.
```js
doSignalingHandshake(pc1, pc2, {modfiyOffer: offer => myRegExp(offer)});
```
The above produces a false positive, due to my typo (did you catch it?), a problem the spec API does not have:
```js
await pc1.setLocalDescription(myRegExp(await pc1.createOffer()));
await pc2.setRemoteDescription(pc1.localDescription);
await pc2.setLocalDescription(await pc2.createAnswer());
await pc1.setRemoteDescription(pc2.localDescription);
```
The above is appropriate for the (few) who need to mangle SDP IMHO. If not, I'd prefer the more typo-safe:
```js
doSignalingHandshakeModifyOfferAnswer(pc1, pc2, offer => myRegExp(offer), answer => answer);
```
As to plain *doSignalingHandshake*, once [crbug 740501](https://bugs.chromium.org/p/chromium/issues/detail?id=740501) is fixed, could we use the spec-provided API instead?